File: Employee.cs

class Employee{

    string name;
    string designation;
    string year;

    Employee(string _name, string _designation, string _year){
        this.name=name;
        this.designation=_designation;
        this.year=_year;
    }

    void CalculateAge(){
        console.log("age");
    }

}

class Main{

    void main(){
        Employee e= new Employee("Alex", "Developer", 1995);
        Employee e2= new Employee("Alex2", "Developer", 1996);
        Employee e3= new Employee("Alex3", "Developer", 1997);
    }
}

File: JavaScript/employee.js


function Employee(name, designation, year) {
    this.name = name;
    this.designation = designation;
    this.year = year;
}

Employee.prototype.calculateAge = function calculateAge () {
    console.log("age of " + this.name + "=" + (2021 - this.year));
}


console.log(Employee.prototype);

let emp1 = new Employee("Alex", "Tester", 1995);
console.log(emp1);
emp1.calculateAge();

let emp2 = new Employee("Annie", "Developer", 1990);
console.log(emp2);
emp2.calculateAge();
